pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -679,60 +687,96 @@ def process_launch(driver: ProcessDriver, relay: EventRelay, args: List[str], db
 ]
 
 
-def process_attach(driver: ProcessDriver, relay: EventRelay, args: List[str], dbg: LLDB) -> None:
+def _attach_with_info(
+    driver: ProcessDriver, relay: EventRelay, dbg: LLDB, info: lldb.SBAttachInfo, cont=False
+):
     """
-    Attaches to a process with the given arguments.
+    Attaches to a process based on SBAttachInfo information
     """
-    args = parse(args, process_attach_ap, process_attach_unsupported)
-    if not args:
-        return
-
     targets = dbg.debugger.GetNumTargets()
     assert targets < 2
     if targets == 0:
         print(message.error("error: no target, create one using the 'target create' command"))
         return
 
+    # TODO/FIXME: This should ask:
+    # 'There is a running process, detach from it and attach?: [Y/n]'
```

Comment:
Suggestion: instead implement an optional general kill/detach mechanism for all commands that can spawn or attach to a process, since we do this same check on all of them.